File: day5/puzzle2.py
import re
c = 0
for i in open('day5.txt'):
 m1 = len(re.findall(r'(.)(.).*\1\2', i))
 m2 = len(re.findall(r'(.)(.)\1', i))
 m = (m1 > 0) and (m2 > 0)
 if m:
  c = c + 1
print c

File: 2015/05/puzzle2.js
var fs = require('fs');

function count(re, s) {
 for (var i = 0; m = re.exec(s); ++i);
 return i;
}

fs.readFile('day5.txt', 'utf-8', function (err, inp) {
 if (err) throw err;
 console.log(inp.trim().split(/\n/g).reduce(function (curr, line) {
  var ok = (count(/(.)(.).*\1\2/g,line) > 0 && count(/(.)(.)\1/g,line) > 0)
  return curr + (ok ? 1 : 0);
 }, 0));
});
